app/modules/site_settings/models.py

```diff
@@ -151,14 +175,25 @@ def validate_custom_fields(value, category_str, field_str):
 #
 # for now, it basically will disallow changing a definition entirely
 def change_custom_field_data(cf_def, field_str):
+    cf_id = cf_def.get('id')
     raise HoustonException(log, f'{field_str} id {cf_id} cannot be changed')
 
+
 # NOTE WIP!
-def custom_field_data(cfd_id, cls_table_name):
+#   this is for potential use by change_custom_field_data() above, to access actual data/values
+#   *in use* by the customField
+def custom_field_data(cf_def, class_name):
+    cf_id = cf_def.get('id')
+    assert cf_id
+    has_values = []
     # h/t https://stackoverflow.com/a/68679549 for this madness
-    res = db.session.execute('SELECT guid FROM annotation WHERE (bounds #>> \'{}\')::jsonb->\'rect\' IS NULL')
+    res = db.session.execute(
+        f'SELECT guid FROM {class_name.lower()} WHERE (custom_fields #>> \'{{}}\')::jsonb->\'{cf_id}\' IS NULL'
+    )
     for r in res:
```

Comment:
meaningful names please. 
res = what ?
r = no really what?

Reply:
i consider this code _very_ provisional.  :grin:  maybe once i fully believe it is useful and doing the right thing, i will get more serious about it. haha